model/src/main/java/org/cbioportal/model/DensityPlotData.java

```diff
@@ -0,0 +1,39 @@
+package org.cbioportal.model;
+
+import javax.validation.constraints.NotNull;
+import java.io.Serializable;
+import java.math.BigDecimal;
+import java.util.List;
+
+public class DensityPlotData implements Serializable {
+    @NotNull
+    private List<DensityPlotBin> bins;
```

Comment:
In most places where we use `@NotNull`, the parent object is never directly instantiated. Instead, it either comes from the API, in which case Jackson makes it, or it comes from the database, in which case MyBatis makes it. This all makes it much easier for the `@NotNull` assertion to always be true.

In your case, you actually use the constructor directly, so you do have DensityPlatData objects that have null bins. I think the best way to address this is to assign an empty arraylist to bins when you declare it. I realize we don't do this elsewhere, but again, we don't instantiate most of our objects directly.

```suggestion
    private List<DensityPlotBin> bins = new ArrayList<>();
```